setup.py

```diff
@@ -82,6 +82,8 @@
         "pandas",
         # Plotting learning curves
         "matplotlib",
+        # for better typing support for users with Python <3.11
```

Comment:
this is not needed anymore, no?
and requirement should be conditioned by python version